core/embed/rust/src/ui/layout_eckhart/flow/continue_recovery_homepage.rs

```diff
@@ -206,94 +184,187 @@ pub fn new_continue_recovery_homepage(
             "abort_recovery".into(),
         ));
 
-    let res = if show_instructions {
-        let content_menu =
-            VerticalMenuScreen::new(VerticalMenu::<ShortMenuVec>::empty().with_item(
-                Button::new_menu_item(cancel_btn.into(), theme::menu_item_title_orange()),
+    match (show_instructions, remaining_shares) {
+        (true, _) => flow_before_shares(content_main, cancel_btn),
+        (false, None) => flow_between_shares_simple(content_main, content_cancel),
+        (false, Some((pages, n))) => {
+            flow_between_shares_advanced(content_main, content_cancel, pages, n, cancel_btn)
+        }
+    }
+}
+
+fn recovery_screen_metadata(
+    recovery_type: RecoveryType,
+    show_instructions: bool,
+    subtext: &Option<TString<'static>>,
+) -> (
+    Header,
+    TString<'static>,
+    TString<'static>,
+    TString<'static>,
+    TString<'static>,
+) {
+    match recovery_type {
```

Comment:
Personally, I am also fine with keeping this logic as part of `new_continue_recovery_homepage`, since IMO it's a bit more readable to keep the following `let` "closer" to the `match` statement:
```rust
let (header, verb, cancel_btn, cancel_title, cancel_intro) = match recovery_type {
...
};
```

Feel free to resolve in case of disagreement :)